### PR TITLE
ARROW-15080: [Python][C++] Enable tuples conversion to interval

### PR DESCRIPTION
**Pull request by emkornfield**
Allows tuples to be converted to MonthDayNanoInterval
when type is explicitly specified.

Also, add missing branch for intervals to strategy.py

### CODE REVIEW

**Comment by emkornfield** [comment]
CI failure looks unrelated (TestS3FSGeneric.DeleteRootDirContents on ARM)

**Comment by emkornfield** [comment]
CC @pitrou

File: cpp/src/arrow/python/python_to_arrow.cc [hunk 1]
```diff
@@ -416,8 +416,18 @@ class PyValue {
...
+                                               "Months (Index 0) index to large"));
+        RETURN_NOT_OK(internal::CIntFromPython(PyTuple_GET_ITEM(obj, 1), &output.days,
+                                               "Days (Index 1) index to large"));
+        RETURN_NOT_OK(internal::CIntFromPython(PyTuple_GET_ITEM(obj, 2),
+                                               &output.nanoseconds,
+                                               "Nanoseconds (Index 2) index to large"));
+        return output;
+      }
+
       // date_offset can have zero fields.
```
Comment (pitrou) [inline cpp/src/arrow/python/python_to_arrow.cc:431]: Move this comment before the enclosing `if`?

File: cpp/src/arrow/python/python_to_arrow.cc [hunk 2]
```diff
@@ -416,8 +416,18 @@ class PyValue {
...
-
     if (ARROW_PREDICT_FALSE(!found_attrs) && !is_date_offset) {
+      if (PyTuple_Check(obj) && PyTuple_Size(obj) == 3) {
+        RETURN_NOT_OK(internal::CIntFromPython(PyTuple_GET_ITEM(obj, 0), &output.months,
+                                               "Months (Index 0) index to large"));
+        RETURN_NOT_OK(internal::CIntFromPython(PyTuple_GET_ITEM(obj, 1), &output.days,
+                                               "Days (Index 1) index to large"));
+        RETURN_NOT_OK(internal::CIntFromPython(PyTuple_GET_ITEM(obj, 2),
+                                               &output.nanoseconds,
+                                               "Nanoseconds (Index 2) index to large"));
```
Comment (pitrou) [inline cpp/src/arrow/python/python_to_arrow.cc:427]: Wording suggestion:
```suggestion
        RETURN_NOT_OK(internal::CIntFromPython(PyTuple_GET_ITEM(obj, 0), &output.months,
                                               "Months (tuple item #0) too large"));
        RETURN_NOT_OK(internal::CIntFromPython(PyTuple_GET_ITEM(obj, 1), &output.days,
                                               "Days (tuple item #1) too large"));
        RETURN_NOT_OK(internal::CIntFromPython(PyTuple_GET_ITEM(obj, 2),
                                               &output.nanoseconds,
                                               "Nanoseconds (tuple item #2) too large"));
```

**Review by pitrou — COMMENTED** [review]
Thanks @emkornfield. Just two minor suggestions.

File: cpp/src/arrow/python/python_to_arrow.cc (hunk 1 above)
Comment (emkornfield) [reply cpp/src/arrow/python/python_to_arrow.cc:431]: yep

**Comment by emkornfield** [comment]
@github-actions autotune everything

**Review by pitrou — APPROVED** [review]
LGTM, thank you @emkornfield

**Pull request closed without merging**